src/data/set/basic.lean

```diff
@@ -132,7 +137,34 @@ assume h : x ∈ ∅, h
 @[simp] theorem not_not_mem [decidable (a ∈ s)] : ¬ (a ∉ s) ↔ a ∈ s :=
 not_not
 
-/- empty set -/
+/-! ### Non-empty sets -/
+
+/-- A set `s` is not empty. -/
+protected def nonempty (s : set α) : Prop := ∃ x, x ∈ s
+
+lemma nonempty.of_mem {x} (h : x ∈ s) : s.nonempty := ⟨x, h⟩
```

Comment:
Why not `nonempty_of_mem` here?

Reply:
Probably I should just remove this function. It's always easier to write `⟨_, h⟩`.

Reply:
I think it's worth keeping. `⟨_, h⟩` will only elaborate if the expected type is known.

Reply:
Renamed.